### PR TITLE
Update to v0.3.0

### PR DESCRIPTION
**Pull request by petreeftime**
Signed-off-by: Petre Eftime <epetre@amazon.com>

### Summary of the PR

*Please summarize here why the changes in this PR are needed.*

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+## Changes
+- [[#40]](https://github.com/rust-vmm/seccompiler/pull/40): Update Rust
+  to Edition 2021
+
+## Fixes
+
+- [[#31]](https://github.com/rust-vmm/seccompiler/issues/31): Implement
+  From<BackendError> for Error
+- [[#40]](https://github.com/rust-vmm/seccompiler/pull/40): Fix clippy
+  complaints about missing Eq when PartialEq is implemented
```
Comment (aghecenco) [inline CHANGELOG.md:12]: nit: enclose `From<BackendError> for Error`, `Eq` and `PartialEq` in code backtics so they show up as code

**Review by aghecenco — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
+# v0.3.0
+
+## Changes
```
Comment (lauralt) [inline CHANGELOG.md:3]: The convention in rust-vmm is to use Changed, Fixed.. instead of Changes, Fixes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,16 @@
+# v0.3.0
+
+## Changed
+- [[#40]](https://github.com/rust-vmm/seccompiler/pull/40): Update Rust
+  to Edition 2021
+
+## Fixed
+
+- [[#31]](https://github.com/rust-vmm/seccompiler/issues/31): Implement
```
Comment (lauralt) [inline CHANGELOG.md:9]: I think this would fit better in `Added` instead of `Fixed` :-?

**Review by lauralt — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (petreeftime) [reply CHANGELOG.md:9]: I think it's better as a fix, as it made error-handling very ugly without.

**Review by axlprv — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**